docs/en/tutorials/getting-started.rst

```diff
@@ -228,33 +227,196 @@ entity definition:
+            {
+                $this->username = $username;
+            }
+
+            public function getPasswordHash(): string
+            {
+                return $this->passwordHash;
+            }
+
+            public function setPasswordHash(string $passwordHash): void
+            {
+                $this->passwordHash = $passwordHash;
+            }
+
+            public function getBans(): array
+            {
+                return $this->bans;
+            }
+
+            public function addBan(Ban $ban): void
+            {
+                $this->bans[] = $ban;
+            }
         }
 
-        public function getName()
+Here, we avoid all possible logic from the model and only care about injecting
+data into it without validation nor consideration about the object's state.
+
+And as Doctrine ORM is a persistence tool for your domain, the state of an
+object is really important. This is why we recommend using rich models.
+
+**Rich models: Mutators and DTOs**
+
+We recommend using a rich model design and rely on more complex
+mutators, ideally based on DTOs.
+This way, you should **not** use getters nor setters, and instead,
+implement methods that represent the **behavior** of your domain.
+
+For example, when having a ``User`` entity, we could foresee
+the following kind of optimization.
+
+Example of a rich model with proper accessors and mutators:
+
+.. configuration-block::
+
+    .. code-block:: php
+        <?php
+        class User
         {
-            return $this->name;
+            private $banned;
+            private $username;
+            private $passwordHash;
+            private $bans;
+
+            public function toNickname(): string
+            {
+                return $this->username;
+            }
+
+            public function authenticate(string $password, callable $checkHash): bool
+            {
+                return $checkHash($password, $this->passwordHash) && ! $this->hasActiveBans();
+            }
+
+            public function changePass(string $password, callable $hash): void
+            {
+                $this->passwordHash = $hash($password);
+            }
+
+            public function ban(\DateInterval $duration): void
+            {
+                assert($duration->invert !== 1);
+
+                $this->bans[] = new Ban($this);
+            }
         }
 
-        public function setName($name)
+Additionally, our entities should never see their state change
+partially without validation. For example, creating a ``new Product()``
+object without any data makes it an **invalid object**.
+Rich models should represent **behavior**, not **data**, therefore
+they should be valid even after a ``__construct()``.
+
+To help creating such objects, we can rely on ``DTO``s, and/or make
+our entities always up-to-date. This can be performed with static constructors,
+or rich mutators that accept the ``DTO`` as argument. And DTOs can be
```

Comment:
s/the DTO as argument/DTOs as arguments

Reply:
s/arguments/parameters

Reply:
Also here: sentence started with `And`: don't do that. Split over a new paragraph and explain that a DTO maintains some internal state consistency that we can rely upon.